app/src/main/java/de/tum/in/tumcampusapp/component/ui/studyroom/StudyRoomsActivity.java

```diff
@@ -168,32 +150,61 @@ public void openLink(View view) {
         startActivity(findStudyRoomIntent);
     }
 
-    @Override
-    protected Void onLoadInBackground(Void... arg) {
-        StudyRoomGroupManager sm = new StudyRoomGroupManager(this);
-        try {
-            sm.downloadFromExternal();
-        } catch (JSONException e) {
-            Utils.log(e);
-            // No error handling here
-        }
-        return null;
+    private void loadStudyRooms() {
+        showLoadingStart();
+        TUMCabeClient
+                .getInstance(this)
+                .getStudyRoomGroups(new Callback<List<StudyRoomGroup>>() {
+                    @Override
+                    public void onResponse(@NonNull Call<List<StudyRoomGroup>> call,
+                                           @NonNull Response<List<StudyRoomGroup>> response) {
+                        List<StudyRoomGroup> groups = response.body();
+                        if (response.isSuccessful() && groups != null && !groups.isEmpty()) {
+                            onDownloadSuccessful(groups);
+                        } else if (response.isSuccessful()) {
+                            showError(R.string.error_no_data_to_show);
+                        } else {
+                            showErrorLayout();
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(@NonNull Call<List<StudyRoomGroup>> call,
+                                          @NonNull Throwable t) {
+                        Utils.log(t);
+                        if (t instanceof NoNetworkConnectionException) {
+                            showNoInternetLayout();
+                        } else {
+                            showErrorLayout();
+                        }
+                    }
+                });
     }
 
-    @Override
-    protected void onLoadFinished(Void result) {
+    private void onDownloadSuccessful(@NonNull List<StudyRoomGroup> groups) {
+        Handler handler = new Handler(Looper.getMainLooper());
+        handler.post(() -> {
+            updateDatabase(groups);
+            runOnUiThread(() -> displayStudyRooms(groups));
```

Comment:
This doesn't make sense :thinking:.
`runOnUiThread` is the main looper and updating the database probably shouldn't happen on the main thread